src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -8,55 +8,30 @@ public GildedRose(Item[] items) {
     }
 
     public void updateQuality() {
-        for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
-            } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
-
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-
-                        if (items[i].sellIn < 6) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
-                    }
-                }
-            }
-
-            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                items[i].sellIn = items[i].sellIn - 1;
-            }
-
-            if (items[i].sellIn < 0) {
-                if (!items[i].name.equals("Aged Brie")) {
-                    if (!items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].quality > 0) {
-                            if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                                items[i].quality = items[i].quality - 1;
-                            }
-                        }
-                    } else {
-                        items[i].quality = items[i].quality - items[i].quality;
-                    }
-                } else {
-                    if (items[i].quality < 50) {
-                        items[i].quality = items[i].quality + 1;
-                    }
-                }
+        for (Item item : items) {
```

Comment:
I'm curious if this can just be just:
```Java
for (Item item : items) {
  item.age();
}

I know that we can't modify the `Item` class at all but you could wrap Item in another class.
```

Reply:
We were thinking about doing this near the beginning but felt as though it was kind of working around the requirement the client set in a way that could upset the goblin and we really didn't want to get one shotted, that being said we are not opposed to the idea in general. My other thought was going the route of creating a hashmap wrapper (called handlers or something) that implemented the builder pattern, whose creation and population exists on the app level (so devs could easily access and interact with it).
The hashmap would contain a string key (the names of the items) and the value would be the corresponding way each items quality/sellIn modification are handled via a `Consumer Interface`. A `getOrDefault()` could be used so that if the name does not exist in the hashmap wrapper, the defaultitem consumer interface function would be called.

Here is a photo to help understand my blob of words:
![image](https://user-images.githubusercontent.com/99982874/179302979-e80a7b6d-f6d5-4970-b63c-098e092dc8dc.png)
And an example `Consumer Interface` Method (Aged Brie):
![image](https://user-images.githubusercontent.com/99982874/179303096-67c0de85-9306-45dd-b6bc-54de45f0d947.png)

However I think your idea/comment is much simpler and better if we are allowed to implement it! What are everyones thoughts on this?
Also if anything was unclear in my explanation please let me know 😄 thank you for help!

Reply:
Can we talk through this some time on Monday? Im also confused by this approach because the way I see it we would still need the `Item` class to implement this `ageable` interface? I might be following you but it might be easier to talk through.